pkg/api/rules.go

```diff
@@ -0,0 +1,205 @@
+// some dependency in api/v1 package. Thus, copying them here avoids the enum issue.
+
+// RuleDiscovery has info for all rules
+type RuleDiscovery struct {
+	RuleGroups []*RuleGroup `json:"groups"`
+}
+
+// RuleGroup has info for rules which are part of a group
+type RuleGroup struct {
+	Name string `json:"name"`
+	File string `json:"file"`
+	// In order to preserve rule ordering, while exposing type (alerting or recording)
+	// specific properties, both alerting and recording rules are exposed in the
+	// same array.
+	Rules          []Rule    `json:"rules"`
+	Interval       float64   `json:"interval"`
+	Limit          int       `json:"limit"`
+	EvaluationTime float64   `json:"evaluationTime"`
+	LastEvaluation time.Time `json:"lastEvaluation"`
+}
+
+type Rule interface{}
+
+type AlertingRule struct {
+	// State can be "pending", "firing", "inactive".
+	State          string                `json:"state"`
+	Name           string                `json:"name"`
+	Query          string                `json:"query"`
+	Duration       float64               `json:"duration"`
+	Labels         labels.Labels         `json:"labels"`
+	Annotations    labels.Labels         `json:"annotations"`
+	Alerts         []*Alert              `json:"alerts"`
+	Health         prom_rules.RuleHealth `json:"health"`
+	LastError      string                `json:"lastError,omitempty"`
+	EvaluationTime float64               `json:"evaluationTime"`
+	LastEvaluation time.Time             `json:"lastEvaluation"`
+	// Type of an alertingRule is always "alerting".
+	Type string `json:"type"`
+}
+
+type RecordingRule struct {
+	Name           string                `json:"name"`
+	Query          string                `json:"query"`
+	Labels         labels.Labels         `json:"labels,omitempty"`
+	Health         prom_rules.RuleHealth `json:"health"`
+	LastError      string                `json:"lastError,omitempty"`
+	EvaluationTime float64               `json:"evaluationTime"`
+	LastEvaluation time.Time             `json:"lastEvaluation"`
+	// Type of a recordingRule is always "recording".
+	Type string `json:"type"`
+}
+
+// AlertDiscovery has info for all active alerts.
+type AlertDiscovery struct {
+	Alerts []*Alert `json:"alerts"`
+}
+
+// Alert has info for an alert.
+type Alert struct {
+	Labels      labels.Labels `json:"labels"`
+	Annotations labels.Labels `json:"annotations"`
+	State       string        `json:"state"`
+	ActiveAt    *time.Time    `json:"activeAt,omitempty"`
+	Value       string        `json:"value"`
+}
+
+func rulesHandler(apiConf *Config, updateMetrics func(handler, code string, duration float64)) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		statusCode := "400"
+		begin := time.Now()
+		defer func() {
+			updateMetrics("/api/v1/rules", statusCode, time.Since(begin).Seconds())
+		}()
+
+		typ := strings.ToLower(r.URL.Query().Get("type"))
```

Comment:
```suggestion
		type := strings.ToLower(r.URL.Query().Get("type"))
```

Reply:
Sorry, but type is a keyword :)

Reply:
right :facepalm: than maybe `queryType`?

Reply:
Done